Automattic/woocommerce-payments
# BNPL: Add Affirm and Afterpay to settings

Fixes #6271 

#### Changes proposed in this Pull Request

* Display options of Affirm  and/or Afterpay/ Clearpay payment options for merchants from eligible countries and currency
* The option should have a corresponding checkbox that indicates as well as allows changing of enabled status of these BNPL payment methods
* Checking or unchecking the boxes should fire the respective event handlers and the status should persist. 

#### Testing instructions

##### Local server preparation
* Check in to any test account of an eligible country. Eg. US - `acct_1LqHU4QocNIKfAAZ` . 

* Add dummy BNPL related fees to the local setup with these commands 
```
wp wcpay add_new_fee base base-us-affirm-fee usd --percentage_rate=0.042 --fixed_rate=42 --countries=US --payment_methods=affirm --mode=test --description="Affirm test"
wp wcpay add_new_fee base base-us-afterpay-fee usd --percentage_rate=0.042 --fixed_rate=42 --countries=US --payment_methods=afterpay_clearpay --mode=test --description="Afterpay test"
```
Add these for any country that you wish to test with by modifying the command accordingly 
* Set the BNPL flag to true
`wp wcpay set_account_meta acct_1LqHU4QocNIKfAAZ --key=bnpl_affirm_afterpay_enabled --value=1`

Make sure you use your test account number in the command.

##### Test 
* Make sure the correct account is showing up in WC Pay Dev tools. Else use CLI to link the desired test account
* Browse to `Payments > Settings` and check the `Payments accepted on checkout` section. You should see Affirm and Afterpay listed with checked out checkboxes
* Monitor Console and React components in browser tools. 
* Click each of the payment methods and save. There should be no errors. and They should feature within the Enabled PM's 
* Refresh the page, you should still see them checked
* Uncheck and save them and it should get removed from the enabled Pms, and the boxes should sustain as unchecked. 

Automated test - The existing tests seem adequate, but if you feel there are some good scenarios being missed, kindly do drop a hint. 

-------------------


**Post merge**
